nextjs/src/app/reports/[id]/page.tsx

```diff
@@ -93,84 +94,92 @@ export default function Page({ params: { id } }: { params: Params }) {
-                        secondIn2019="Conservative"
-                        mpName="Zarah Sultana"
-                        mpImgUrl="https://www.localintelligencehub.com/media/person/mp_4786_7qDOwxw.jpeg"
-                      />
-                      <ReportsConsItem
-                        consName="Bury North"
-                        firstIn2019="Conservative"
-                        secondIn2019="Labour"
-                        mpName="James Daly"
-                        mpImgUrl="https://www.localintelligencehub.com/media/person/mp_4854_BxRRx9j.jpeg"
-                      />
-                      <ReportsConsItem
-                        consName="Camberwell and Peckham"
-                        firstIn2019="Labour"
-                        secondIn2019="Conservative"
-                        mpName="Harriet Harman"
-                        mpImgUrl="https://www.localintelligencehub.com/media/person/mp_150_rgMOVq7.jpeg"
-                      />
-                    </TabsContent>
-                    <TabsContent value="selected-cons-1">Change your password here.</TabsContent>
-                  </Tabs>
+        {report.loading && !report.data?.mapReport ? (
+          <div className="absolute w-full h-full inset-0">
+            <div className="flex flex-col items-center justify-center w-full h-full">
+              <LoadingIcon />
+            </div>
+          </div>
+        ) : (
+          <div className="absolute top-5  left-5 right-0 w-0">
+            <div className="flex flex-col items-start gap-4">
+              <Card className="w-[200px] p-4 bg-white border-1 border-meepGray-700 text-meepGray-800">
+                <CardHeader className="flex flex-row items-center mb-4">
+                  <CardTitle contentEditable id="nickname" className="text-hMd grow font-IBMPlexSansMedium" onBlur={d => {
+                    updateMutation({
+                      name: document.getElementById("nickname")?.textContent?.trim()
+                    })
+                  }}>
+                    {report.data?.mapReport.name}
+                  </CardTitle>
+                  <DropdownMenu>
+                    <DropdownMenuTrigger>
+                      <MoreVertical className='w-3' />
+                    </DropdownMenuTrigger>
+                    <DropdownMenuContent side="right" align="start">
+                      <DropdownMenuLabel>Report Settings</DropdownMenuLabel>
+                      <DropdownMenuSeparator />
+                      <DropdownMenuItem>Share</DropdownMenuItem>
+                      <DropdownMenuItem>Invite</DropdownMenuItem>
+                      <DropdownMenuItem onClick={() => setDeleteOpen(true)}>Delete</DropdownMenuItem>
+                    </DropdownMenuContent>
+                  </DropdownMenu>
                 </CardHeader>
+                <CardContent>
+                  <ToggleGroup type="multiple" variant="outline">
+                    {/* @ts-ignore */}
```

Comment:
While I think this is fine to do, is it an issue with library typing or an issue with us doing something unorthodox?

Reply:
It's an issue with the useFragment hook which recursively questions every property's existence.

I'm thinking we could replace these typecheck bail-outs with a runtime Zod parser, perhaps auto-generated out of the graphql queries.